src/pollen.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Pollen charm business logic."""
+
+import os
+import shutil
+import time
+from glob import glob
+from subprocess import run
+
+from charms.operator_libs_linux.v0 import apt
+from charms.operator_libs_linux.v1 import systemd
+
+from exceptions import ConfigurationWriteError, InstallError
+
+
+def prepare() -> None:
+    """Install packages and write configuration files.
+
+    Raises:
+        InstallError: if the packages fail to install
+        ConfigurationWriteError: something went wrong writing the configuration
+    """
+    try:
+        apt.update()
+        apt.add_package(["pollen", "pollinate", "ent"])
+        # Apt-get makes the pollen restart error if we don't wait,
+        # even if apt-get executes correctly and there are no
+        # child processes left. The cause is unknown.
+        time.sleep(10)
```

Comment:
I'm wondering if we shouldn't isolate the install of pollen, meaning install `ent` and `pollinate` then install `pollen` and wait (if that's still needed)

Reply:
I can try that to see if it changes anything, but if it doesn't what would be the purpose of isolating it?

Reply:
If that doesn't solve the sleep issue, indeed it's not worth it